### ISSUE TITLE
Provide alternative to mustache templates

### ISSUE THREAD

**Issue by DaSchTour** [FACT-Finder-Web-Components/ff-web-components#19]
When using fact finder web components inside an angular application the mustache template conflicts with angular templates. It would be great if there was an alternative possibility for templating, for example by using attributes like `<span data-bind="record.Name"></span>`

**Comment by toarm** [comment]
Hi @DaSchTour ,
this is currently a more complicated task we cant achieve within a reasonable amount of time unfortunately. 
I'm not familiar with angular but is there a possibility of writing something like an "angular adapter" which will do the mapping? 
If you can dip my nose into the right direction (angular docs or similar), we might be able to figure something out...

**Comment by DaSchTour** [comment]
I'm not sure about an angular adapter. The problem is, that all `{{foo}}` elements are interpreted by angular. And if these properties are not provided by the controller compiling the templates failed and the app can't be build. I tried with different approaches like inserting the properties through an angular pipe that wraps it into `{{}}`but that didn't work very stable, because it seams to be rendered at a later time so the components sometimes didn't pick it up correctly. I also tried to insert templates through angulars innerHTML property but that also didn't work very stable.

Maybe some kind of adapter could be build with [structural directive](https://angular.io/guide/structural-directives) and `<ng-template>`by providing the data for a component as context to angular. But I don't really have an idea how that might work.

**Comment by toarm** [comment]
mh, that doesnt sound like a feature request anymore :/ 

In this case I'm pretty sure @KrzysztofKozlowskiOBSS knows more, but he's on vacation until Wednesday.

**Comment by mwinkelm** [comment]
No idea if this might work, but what if we could optionally configure **[custom delimiters](https://github.com/janl/mustache.js#custom-delimiters)** at `<ff-communication>` which are passed to the mustache renderer?

**Comment by toarm** [comment]
> No idea if this might work, but what if we could optionally configure **[custom delimiters](https://github.com/janl/mustache.js#custom-delimiters)** at `<ff-communication>` which are passed to the mustache renderer?

Sounds very reasonable to me and definetly worth a try. We would have to upgrade to latest mustache version but that shouldnt be a big deal I guess.

**Comment by toarm** [comment]
https://share.omikron.net/index.php/s/sVwSaxHul3mOXLu
pw: mustache (I cant provide url without one 🤷‍♂ )
It's a in-dev SNAPSHOT version. Tests are passing but no final QA made, use carefully. Link expires end of july. 
```html
        <script>
            document.addEventListener("ffReady", function () {
                factfinder.libs.mustache.tags = ["[", "]"]
            });
        </script>
        <script src="../../../dist/vendor/custom-elements-es5-adapter.js"></script>
        <script src="../../../dist/vendor/webcomponents-loader.js"></script>
        <script defer src="../../../dist/bundle.js"></script>
```

edit: 
FYI #20 is not included in that snapshot.

**Comment by toarm** [comment]
https://share.omikron.net/index.php/s/fY79reTF5UyjKPj -> #20 included
pw same as above

**Comment by mwinkelm** [comment]
**That works!** If you can manage it that all components use that custom delimiter within the templates this might be a good solution to overcome specific framework syntax race conditions.

Currently I have the following setup within our Angular application:

```
<ff-asn>
  <ff-asn-group>
	<div slot="groupCaption" class="groupCaption">
		[[group.name]]
		<span class="down">+</span>
	</div>
	<div data-container="detailedLinks">
		<div data-content="detailedLinks"></div>
	</div>
	<div data-container="showMore">
		<span class="text">See all</span>
	</div>
	<div data-container="hiddenLinks">
		<div data-content="hiddenLinks"></div>
	</div>
	<div data-container="showLess">
		<span class="text">See less</span>
	</div>
  </ff-asn-group> 
</ff-asn>
```
Which produces the following result with `[[` and `]]` as delimiter.

![filters](https://user-images.githubusercontent.com/6881829/60722649-d6f36500-9f31-11e9-933f-f669981ba410.png)

As you can see, the `ff-asn-group-slider` is still the default and uses `{{` and `}}` as delimiters.

**Comment by toarm** [comment]
> As you can see, the ff-asn-group-slider is still the default and uses {{ and }} as delimiters.

Good catch!  I'll note that in the ticket.

**Comment by toarm** [comment]
Might be good to note that in our upcoming documentation FAQ @wdutkiewicz-obj 
Maybe you want add a placeholder with a link to this thread or so...

**Comment by mwinkelm** [comment]
> Sounds very reasonable to me and definetly worth a try.

Thank you for this blazing fast support!

**Comment by toarm** [comment]
FFWEB-1295 for reference

**Comment by mwinkelm** [comment]
@toarm: Is this already included in the 3.5.0 release? Looks like it is working with that version.

**Comment by ghost** [comment]
We are still working on it. Currently we are updating all our internal default templates to adjust to dynamic Mustache delimiter. We are looking forward to a full implementation soon and we will close this ticket as soon as we think we are ready. 

Thanks for your nice feature request and feedback! :)

**Comment by toarm** [comment]
Hi @mwinkelm @DaSchTour ,
we would love to get your feedback before releasing this feature in a new version.

https://github.com/FACT-Finder-Web-Components/ff-web-components/releases/tag/3.5.0-configurable-mustache-delimiters
```html
<ff-communication url="https://search-web-components.fact-finder.de/FACT-Finder-7.3"
                  version="7.3"
                  channel="bergfreunde-co-uk"
                  mustache-delimiters="[[,]]">
</ff-communication>
```

**Comment by mwinkelm** [comment]
@toarm Thanks to you and your team for the effort spent into this issue.

@DaSchTour Maybe you have some time left to test whether this prerelease works with the config on `<ff-communication>`? Due to my stay in hospital I can‘t give an immediate feedback currently.

**Comment by DaSchTour** [comment]
Hi @toarm works great. I checked with `<ff-asn>` where we already used this and also with `<ff-record>` and `<ff-breadcrumb-trail>` (which did not work with custom delimiter before) and they work all fine. Thanks a lot

**Comment by toarm** [comment]
> @toarm Thanks to you and your team for the effort spent into this issue.
> 
> @DaSchTour Maybe you have some time left to test whether this prerelease works with the config on `<ff-communication>`? Due to my stay in hospital I can‘t give an immediate feedback currently.

get well soon.

**Comment by omikron-fp** [comment]
Feature is merged into master. Release will probably happen next week.

**Comment by omikron-fp** [comment]
Released with [3.6.0](https://github.com/FACT-Finder-Web-Components/ff-web-components/releases/tag/3.6.0).
Thank you for your cooperation!
